### PR TITLE
linting: use the GIT_BASE_COMMIT env variable

### PR DESCRIPTION
**Pull request by v1v**
## What does this pull request do?

Use `GIT_BASE_COMMIT` since it's the variable with the current commit therefore all the integrations should keep working as is.

### CODE REVIEW

File: .ci/linting.groovy [hunk 1]
```diff
@@ -19,16 +20,36 @@ pipeline {
...
+  try {
+    gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: false)
+  } catch (err) {
+    // NOOP: avoid failing if non-elasticians, this will avoid issues when PRs comming
+    //       from non elasticians since the validation will not fail
+  }
+}
+
+def runPreCommit() {
+  docker.image('python:3.7-stretch').inside(){
```
Comment (beniwohli) [inline .ci/linting.groovy:51]: While we're at it, we could update this to `python:3.10-bullseye`. `stretch` is EOL, and while Python 3.7 isn't for another year, some of the tooling we use in the linting stage might drop support for it earlier than that.

However, we can also do that in a separate PR if you prefer.

```suggestion
  docker.image('python:3.10-bullseye').inside(){

**Review by beniwohli — APPROVED** [review]

File: .ci/linting.groovy (hunk 1 above)
Comment (v1v) [reply .ci/linting.groovy]: ``
13:16:57  + curl -s https://pre-commit.com/install-local.py
13:16:57  + python3 -
13:16:57  <stdin>:15: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
13:16:58  /tmp/.virtualenv-pkg/virtualenv.py:24: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
13:16:58    import distutils.spawn
13:16:58  /tmp/.virtualenv-pkg/virtualenv.py:25: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
13:16:58    import distutils.sysconfig
13:16:58  Using base prefix '/usr/local'
13:16:58  New python executable in /var/lib/jenkins/workspace/agent-python-linting-mbp_PR-1542/src/.pre-commit-venv/bin/python3
13:16:58  Also creating executable in /var/lib/jenkins/workspace/agent-python-linting-mbp_PR-1542/src/.pre-commit-venv/bin/python
13:16:58  Traceback (most recent call last):
13:16:58    File "/tmp/.virtualenv-pkg/virtualenv.py", line 2633, in <module>
13:16:58      main()
13:16:58    File "/tmp/.virtualenv-pkg/virtualenv.py", line 859, in main
13:16:58      create_environment(
13:16:58    File "/tmp/.virtualenv-pkg/virtualenv.py", line 1161, in create_environment
13:16:58      install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages=site_packages, clear=clear, symlink=symlink)
13:16:58    File "/tmp/.virtualenv-pkg/virtualenv.py", line 1720, in install_python
13:16:58      fix_local_scheme(home_dir, symlink)
13:16:58    File "/tmp/.virtualenv-pkg/virtualenv.py", line 1806, in fix_local_scheme
13:16:58      if sysconfig._get_default_scheme() == "posix_local":
13:16:58  AttributeError: module 'sysconfig' has no attribute '_get_default_scheme'. Did you mean: 'get_default_scheme'?
```

It didn't like it the python 3.10 :/

File: .ci/linting.groovy [hunk 2]
```diff
@@ -19,16 +20,36 @@ pipeline {
...
+  try {
+    gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: false)
+  } catch (err) {
+    // NOOP: avoid failing if non-elasticians, this will avoid issues when PRs comming
+    //       from non elasticians since the validation will not fail
+  }
+}
+
+def runPreCommit() {
+  docker.image('python:3.10-bullseye').inside(){
```
Comment (v1v) [inline .ci/linting.groovy:51]:
```suggestion
  docker.image('python:3.10-stretch').inside(){
```

File: .ci/linting.groovy (hunk 1 above)
Comment (v1v) [reply .ci/linting.groovy:51]: Thanks Beni, I'll keep track then we don't forget this conversation
